### PR TITLE
Add .livemd as a markdown extension

### PR DESCRIPTION
**Pull request by contra-bit**
## Needs and benefits
[Livebook](https://livebook.dev/) notebooks are used for code documentation and for deep dives and note-taking in the elixir ecosystem. Rendering these in these as Markdown on frogejo has many benefits, since livemd is a subset of markdown. Some of the benefits are:
- New users of elixir and livebook are scared by unformated .livemd files, but are shown what they expect
- Sharing a notebook is as easy as sharing a link, no need to install the software in order to see the results.

[goldmark-meraid ](https://github.com/abhinav/goldmark-mermaid) is a mermaid-js parser already included in gitea. This makes the .livemd rendering integration feature complete. With this PR class diagrams, ER Diagrams, flow charts and much more will be rendered perfectly.

With the additional functionality gitea will be an ideal tool for sharing resources with fellow software engineers working in the elixir ecosystem. Allowing the git forge to be used without needing to install any software.

## Feature Description
This issue requests the .livemd extension to be added as a Markdown language extension.

- `.livemd` is the extension of Livebook which is an Elixir version of Jupyter Notebook.
- `.livemd` is` a subset of Markdown.

This would require the .livemd to be recognized as a markdown file. The Goldmark the markdown parser should handle the parsing and rendering automatically.

Here is the corresponding commit for GitHub linguist:
https://github.com/github/linguist/pull/5672

Here is a sample page of a livemd file:
https://github.com/github/linguist/blob/master/samples/Markdown/livebook.livemd

## Screenshots

The first screenshot shows how github shows the sample .livemd in the browser.
The second screenshot shows how mermaid js, renders my development notebook and its corresponding ER Diagram. The source code can be found here: https://codeberg.org/lgh/Termi/src/commit/79615f74281789a1f2967b57bad0c67c356cef1f/termiNotes.livemd

## Testing
I just changed the file extension from `.livemd`to `.md`and the document already renders perfectly on codeberg. Check you can it out [here](https://codeberg.org/lgh/Termi/src/branch/livemd2md/termiNotes.md)

### CODE REVIEW

File: modules/setting/setting.go [hunk 1]
```diff
@@ -348,7 +348,7 @@ var (
 	}{
 		EnableHardLineBreakInComments:  true,
 		EnableHardLineBreakInDocuments: false,
-		FileExtensions:                 strings.Split(".md,.markdown,.mdown,.mkd", ","),
+		FileExtensions:                 strings.Split(".md,.markdown,.mdown,.mkd,.livemd", ","),
```
Comment (delvh) [inline modules/setting/setting.go]: Please change it in `custom/conf/app.example.ini#1318` and `docs/content/doc/advanced/config-cheat-sheet.en-us.md` too.

Comment (delvh) [reply modules/setting/setting.go]: Huh? That whole paragraph is missing in the cheat sheet?
How did that happen?
Would you mind adding it now?

Comment (contra-bit) [reply modules/setting/setting.go]: I have added the new .extension to the docs and example

File: docs/content/doc/advanced/config-cheat-sheet.en-us.md [hunk 2]
```diff
@@ -116,7 +116,7 @@ In addition there is _`StaticRootPath`_ which can be set as a built-in at build
 
 ### Repository - Editor (`repository.editor`)
 
-- `LINE_WRAP_EXTENSIONS`: **.txt,.md,.markdown,.mdown,.mkd,**: List of file extensions for which lines should be wrapped in the Monaco editor. Separate extensions with a comma. To line wrap files without an extension, just put a comma
+- `LINE_WRAP_EXTENSIONS`: **.txt,.md,.markdown,.mdown,.mkd,.livemd,**: List of file extensions for which lines should be wrapped in the Monaco editor. Separate extensions with a comma. To line wrap files without an extension, just put a comma
```
Comment (delvh) [inline docs/content/doc/advanced/config-cheat-sheet.en-us.md]: That is a different setting: `[repository.editor].LINE_WRAP_EXTENSIONS` vs `[markdown].FILE_EXTENSIONS`.
There's rather the question: Should the former setting also be line wrapped?

Comment (contra-bit) [reply docs/content/doc/advanced/config-cheat-sheet.en-us.md]: Yes, line wrapping makes sense. The Livebook also wraps lines. 

I think it also improves the user experience. Thanks for asking

**Comment by silverwind** [comment]
I'm skeptical whether we really need `goldmark-mermaid`, how does it call into the JS library? I see a package.json in the go project. This seems like something that is prone to go out of date as soon as the project is abandoned.

**Review by wxiaoguang — APPROVED** [review]
It's only a config change, so LGTM.

**Comment by silverwind** [comment]
Has conflicts

**Review by delvh — APPROVED** [review]

**Comment by wxiaoguang** [comment]
Maybe maintainers with writer permission could do a quick fix? That typo seems easy to be fixed.

And it can mention crystalcommunication  as co-author in the description.

**Comment by silverwind** [comment]
So it needs a typo fix and a full rebase, I guess I can do that later today.

**Comment by wxiaoguang** [comment]
Hmm, yes, since "This branch has conflicts that must be resolved", so this PR still needs to be updated, the typo could be fixed together.

**Comment by wxiaoguang** [comment]
> Wait, do not remove the extra comma, I think it's done by purpose. It means "no extension"?

Checked, the removal of fdfbd3d03c603e14f9cb57455afeb891a796e0a1 is right. 👍

**Pull request merged**